trio/_core/_run.py

```diff
@@ -1167,7 +1167,10 @@ def __del__(self) -> None:
 @attr.s(eq=False, hash=False, repr=False, slots=True)
 class Task(metaclass=NoPublicConstructor):
     _parent_nursery: Nursery | None = attr.ib()
-    coro: Coroutine[Any, outcome.Outcome[object], Any] = attr.ib()
+    # could be typed as `Coroutine[Any, outcome.Outcome[object], Any]` but
+    # the code does a lot of dynamic introspection on it and as long as it passes
+    # those it's fine.
+    coro: Any = attr.ib()
```

Comment:
I was thinking that this could stay typed as `Coroutine` for the public API of the class, but cast the `coro` local variable in `iter_await_frames()`. That way type checking is still done.